src/app/containers/orders-preview/orders-preview.component.ts

```diff
@@ -17,12 +19,21 @@ export class OrdersPreviewComponent {
   status$ = new BehaviorSubject(DEFAULT_STATUS);
 
   constructor(
-    public orderManager: OrderManagerFacade,
-    private orderFacade: OrderFacade,
-    private customerFacade: CustomerFacade
+    public orderManager: OrderManagerService,
+    orderFacade: OrderFacade,
+    orderItemFacade: OrderItemFacade,
+    customerFacade: CustomerFacade,
+    productFacade: ProductFacade,
+    accountFacade: AccountFacade
   ) {
+    // TODO: Put inside a resolver and add a check to see if entities have been loaded recently? (mostly the latter)
```

Comment:
Agreed, we should look into resolvers. This is not an uncommon problem with ngrx, where data may be loaded multiple times by different components. If we can use resolvers to manage how often we try to load, and only load when necessary, I think that would be a solid improvement.

Reply:
I think the resolver would do really well with the 'lastLoadAction' proposed feature: 

https://github.com/briebug/ngrx-auto-entity/issues/71

It may seem at first like somewhat of a specific addition that may end up bloating the library, but I think the usecase is common enough that it would be really useful. It would give any developer the ability to create a resolver such as the one mentioned above, and perhaps, if it seems warranted, give us the chance to create a generic resolver that could be used practically out of the box.

That said, this is starting to sound an awful lot like a caching system, and it makes me wonder if we could leverage some other already available tech instead of creating our own. Not sure that would be possible at all since it involves the backend, which is pretty much unknown/none of our concern in the context of the library, but something to think about.